extensions-contrib/influxdb-emitter/src/main/java/org/apache/druid/emitter/influxdb/InfluxdbEmitter.java

```diff
@@ -0,0 +1,231 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.influxdb;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.java.util.emitter.core.Emitter;
+import org.apache.druid.java.util.emitter.core.Event;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Pattern;
+
+
+public class InfluxdbEmitter implements Emitter
+{
+
+  private static final Logger log = new Logger(InfluxdbEmitter.class);
+  private HttpClient influxdbClient;
+  private final InfluxdbEmitterConfig influxdbEmitterConfig;
+  private final AtomicBoolean started = new AtomicBoolean(false);
+  private final ScheduledExecutorService exec = Executors.newScheduledThreadPool(1, new ThreadFactoryBuilder()
+      .setDaemon(true)
+      .setNameFormat("InfluxdbEmitter-%s")
+      .build());
+
+  private final ImmutableSet dimensionWhiteList;
+
+  private final LinkedBlockingQueue<ServiceMetricEvent> eventsQueue;
+
+  public InfluxdbEmitter(InfluxdbEmitterConfig influxdbEmitterConfig)
+  {
+    this.influxdbEmitterConfig = influxdbEmitterConfig;
+    this.influxdbClient = HttpClientBuilder.create().build();
+    this.eventsQueue = new LinkedBlockingQueue<>(influxdbEmitterConfig.getMaxQueueSize());
+
+    this.dimensionWhiteList = ImmutableSet.of(
+        "dataSource",
+        "type",
+        "numMetrics",
+        "numDimensions",
+        "threshold",
+        "dimension",
+        "taskType",
+        "taskStatus",
+        "tier"
+    );
+
+    log.info("constructing influxdb emitter");
+  }
+
+  @Override
+  public void start()
+  {
+    synchronized (started) {
+      if (!started.get()) {
+        exec.scheduleAtFixedRate(
+            new ConsumerRunnable(),
```

Comment:
nit: drop ConsumerRunnable and
```suggestion
            () -> transformAndSendToInfluxdb(eventsQueue),
```